The program doesn't work because when we try to start our program by typing

main:start(integer number of customers).

we get the following error

=ERROR REPORT==== 18-Oct-2013::18:39:46 ===
Error in process <0.34.0> with exit value: {badarg,[{orderqueue,order,1,[{file,"orderqueue.erl"},{line,14}]}]}


=ERROR REPORT==== 18-Oct-2013::18:39:46 ===
Error in process <0.35.0> with exit value: {badarg,[{owner,work,1,[{file,"owner.erl"},{line,26}]}]}

We find this odd because if line 14 is a bad argument then line 13 shoud be as well. The same thing goes with the second error report because the error should be thrown for the chef as well.

Can you describe to us what is wrong?